spec/tictactoe_spec.rb

```diff
@@ -7,7 +7,23 @@
   let (:test_ttt) { TicTacToe.new(game_io, player1, player2) }
 
   it "runs the game loop " do
-    allow(test_ttt.game_loop).to receive(:game_loop) { }
-    test_ttt.game_loop
+    expect(test_ttt.game_loop).to receive(:game_loop).once
+    expect(test_ttt).to receive(:play_again?).and_return(false)
+    test_ttt.play_game
+  end
+
+  it "plays the game twice " do
+    expect(test_ttt.game_loop).to receive(:game_loop).twice
+    allow(test_ttt).to receive(:new_game)
+    expect(test_ttt).to receive(:play_again?).and_return(true, false)
```

Comment:
Nice job here. However, I think you should stub the `@io.get_input` call instead of `play_again?`. The reason is if you stub `play_again?`, you aren't testing that method's functionality anywhere—it's stubbed out in every test so the actual implementation (returning `true` with some "yes" response) is never called/tested. By stubbing `get_input` instead, you're basically saying OK, I have some other object (`GameIO`) with its own behavior and that behavior is under test separately. Assuming I get a certain input from that object and method, this class under test here (`TTT`) will behave like so: (insert expectation).

Alternatively to all of the above, you could instantiate your `TicTacToe` object with a custom `MockIO`.